src/coreComponents/constitutive/fluid/DeadOilFluid.cpp

```diff
@@ -0,0 +1,280 @@
+      {
+        std::cout << strs[i] << std::endl;
+      }
+
+      GEOSX_ERROR_IF( (m_phaseTypes[ip] != PhaseType::WATER)
+                      && strs.size() != 3,
+                      "Three columns (pressure, formation volume factor, and viscosity) are expected for oil and gas" );
+      GEOSX_ERROR_IF( (m_phaseTypes[ip] == PhaseType::WATER)
+                      && strs.size() != 4,
+                      "Four columns (pressure, formation volume factor, compressibility, and viscosity) are expected for water" );
+      GEOSX_ERROR_IF( (m_phaseTypes[ip] == PhaseType::WATER)
+                      && counter > 1,
+                      "The water table must contain only one line" );
+
+      if( m_phaseTypes[ip] == PhaseType::WATER )
+      {
+        m_waterRefPressure = stod( strs[0] );
+        m_waterFormationVolFactor = stod( strs[1] );
+        m_waterCompressibility = stod( strs[2] );
+        m_waterViscosity = stod( strs[3] );
+
+        GEOSX_ERROR_IF( m_waterRefPressure <= 0.0,
+                        "DeadOilFluid: a strictly positive value must be provided for: "
+                        << viewKeyStruct::waterRefPressureString() );
+        GEOSX_ERROR_IF( m_waterFormationVolFactor <= 0.0,
+                        "DeadOilFluid: a strictly positive value must be provided for: "
+                        << viewKeyStruct::waterFormationVolumeFactorString() );
+        GEOSX_ERROR_IF( m_waterCompressibility <= 0.0,
+                        "DeadOilFluid: a strictly positive value must be provided for: "
+                        << viewKeyStruct::waterCompressibilityString() );
+        GEOSX_ERROR_IF( m_waterViscosity <= 0.0,
+                        "DeadOilFluid: a strictly positive value must be provided for: "
+                        << viewKeyStruct::waterViscosityString() );
+      }
+      else
+      {
+        pressureCoords[0].emplace_back( stod( strs[0] ) );
+        formationVolFactor.emplace_back( stod( strs[1] ) );
+        viscosity.emplace_back( stod( strs[2] ) );
+      }
+
+      counter++;
+    }
+
+    if( m_phaseTypes[ip] != PhaseType::WATER )
+    {
+      m_hydrocarbonPhaseOrder.emplace_back( LvArray::integerConversion< integer >( ip ) );
+      string const formationVolFactorTableName = (m_phaseTypes[ip] == PhaseType::OIL) ? "PVDO_Bo" : "PVDG_Bg";
+      string const viscosityTableName = (m_phaseTypes[ip] == PhaseType::OIL) ? "PVDO_visco" : "PVDG_viscg";
+      m_formationVolFactorTableNames.emplace_back( formationVolFactorTableName );
+      m_viscosityTableNames.emplace_back( viscosityTableName );
+
+      GEOSX_ERROR_IF( pressureCoords[0].size() <= 1,
+                      "DeadOilFluid: The oil and gas PVT tables must contain at least 2 values" );
+
+      FunctionManager & functionManager = getGlobalState().getFunctionManager();
+      TableFunction & tablePVDX_B =
+        dynamicCast< TableFunction & >( *functionManager.createChild( "TableFunction", formationVolFactorTableName ) );
```

Comment:
This way of consuming input is nice because it's compatible with what PVTPackage does and can consume Eclipse-style tables directly. Personally, I'd love to also be able to specify DO tables directly in the XML using GEOSX's standard table input capabilities.

The way this could work is one would specify either `tableFiles` or both `densityTableNames` and `viscosityTableNames` attribute on the `DeadOilFluid` node. In the latter case it would fetch existing tables from `FunctionManager` rather than creating them. I'm not sure about water properties though, perhaps they would be input using separate real-valued attributes since it's not really a table. Water also breaks the order of `densityTableNames` and `viscosityTableNames` values, and would need to be worked around (either specifying a dummy table name or just allowing these attributes to have 1 fewer entries).

I don't suggest doing it in this PR, as it already is in good shape for merging; just something to consider in future (maybe I should do it myself later).

Reply:
This was the approach that I initially used for the first commit in this branch, so I just revived it. So to summarize we have two options to populate the model: the "Eclipse-style" way and the "TableFunction" way. There is now a Dead-Oil integrated test that defines the native `TableFunction`s for the formation volume factor and the viscosity in the xml (`deadoil_3ph_staircase_3d.xml`)